__tests__/articlesRoutes.tests.js

```diff
@@ -43,7 +43,101 @@ describe("GET /api/articles/:article_id", () => {
       const {
         body: { message },
       } = await request(app).get("/api/articles/string").expect(400);
-      expect(message).toBe("the aticle id specified is not a valid");
+      expect(message).toBe("the articleId specified is not a valid");
+    });
+  });
+});
+
+describe("GET /api/articles", () => {
+  describe("Happy path", () => {
+    it("should return a 200", async () => {
+      await request(app).get("/api/articles").expect(200);
+    });
+    it("should return an array of objects", async () => {
+      const {
+        body: { articles },
+      } = await request(app).get("/api/articles").expect(200);
+      expect(articles).toBeInstanceOf(Array);
+      expect(articles).not.toHaveLength(0);
+    });
+    it("should return a list of objects that match the article schema", async () => {
+      const {
+        body: { articles },
+      } = await request(app).get("/api/articles").expect(200);
+      articles.forEach((article) => {
+        expect(article).toMatchObject({
+          author: expect.any(String),
+          title: expect.any(String),
+          article_id: expect.any(Number),
+          body: expect.any(String),
+          topic: expect.any(String),
+          created_at: expect.any(String || undefined),
+          votes: expect.any(Number),
+          article_img_url: expect.any(String || undefined),
+          comment_count: expect.any(String),
+        });
+      });
+    });
+  });
+  describe("Sad path", () => {
+    it("Should return the error message 'our database does not have a articles table'", async () => {
+      await db.query(`DROP TABLE IF EXISTS articles CASCADE;`);
+      const {
+        body: { message },
+      } = await request(app).get("/api/articles").expect(404);
+      expect(message).toBe("our database does not have a articles table");
+      await runSeed();
+    });
+    it("Should return the error message 'currently no topics in the database'", async () => {
+      await db.query(`TRUNCATE TABLE articles CASCADE;`);
+      const {
+        body: { message },
+      } = await request(app).get("/api/articles").expect(404);
+      expect(message).toBe("the articles table currently contains no articles");
+      await runSeed();
+    });
+  });
+});
+
+describe("GET /api/articles/:articleId/comments", () => {
+  describe("Happy path", () => {
+    it("should return an array of objects", async () => {
+      const {
+        body: { comments },
+      } = await request(app).get("/api/articles/1/comments").expect(200);
+      expect(comments).toBeInstanceOf(Array);
+      expect(comments).not.toHaveLength(0);
```

Comment:
Could we be more specific on the exact length of the response array that we expect based on the data in the testData file?